sql/moz-fx-data-shared-prod/telemetry_derived/main_summary_v4/part1.sql

```diff
@@ -45,6 +45,25 @@ CREATE TEMP FUNCTION extract_popup_notification_stats(h STRING) AS (
   )
 );
 
+WITH extracted_histograms AS (
+  SELECT
+    document_id,
+    client_id,
+    sample_id,
+    mozfun.hist.extract(payload.histograms.ssl_handshake_result) AS ssl_handshake_result,
+    mozfun.hist.extract(
+      payload.histograms.plugins_notification_shown
+    ) AS plugins_notification_shown,
+    mozfun.hist.extract(
+      payload.histograms.plugins_notification_user_action
+    ) AS plugins_notification_user_action
```

Comment:
```suggestion
    mozfun.hist.extract(payload.histograms.ssl_handshake_result) AS ssl_handshake_result,
    mozfun.hist.extract(
      payload.histograms.plugins_notification_shown
    ) AS plugins_notification_shown,
    mozfun.hist.extract(
      payload.histograms.plugins_notification_user_action
    ) AS plugins_notification_user_action,
    *
```

If we add `*` to this CTE, we can avoid the join. The main query would select from `extracted_histograms` rather than `main`, and thus have access to these extra derived fields.

In this case, the WHERE clause only needs to appear in one place, either here or in the main query but not both.